modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc

```diff
@@ -0,0 +1,71 @@
+[id="cluster-logging-configuration-of-json-log-data-for-default-elasticsearch_{context}"]
+= Configuration of JSON log data for default Elasticsearch
+
+With the default Elasticsearch log store, you must create an index for each format if the JSON log entries _have different formats_.
+
+[IMPORTANT]
+====
+You must create a separate index for each different JSON log format. Otherwise, forwarding different formats to the same index can cause type conflicts and cardinality problems.
+====
+
+To provide a different index for each format, you configure the `ClusterLogForwarder` custom resource. You use a structure type to construct the index name from a category, Kubernetes label, or input selector.
+
+.Structure types
+
+The `ClusterLogForwarder` provides the following structure types for use with the default Elasticsearch log store:
+
+* `structuredTypeName`: (string, optional) If `structuredTypeKey` is not set or its key is not present, use the value of `structuredTypeName` as the structured type.
+* `structuredTypeKey`: (string, optional) If either of the following meta-data keys is set and its value is non-empty, use the value of the key as the structured type.
+** `kubernetes.labels.<key>`: Use the string value of the `kubernetes` label that has the specified `<key>`.
+** `openshift.labels.<key>`: Use the string value of the `openshift` label that has the specified `<key>`.
+
```

Comment:
And  kubernetes.container_name. @Christian Heidenreich  want to introduce this value to customer.
--


<br class="Apple-interchange-newline">

Reply:
@alanconway Is `kubernetes.container_name` a valid value for the `structuredTypeKey` parameter? How does one use it? Does `container_name` function as the key such that OpenShift Logging generates indices that are based on the container names? Or is there more to it?

Reply:
Yes. `kubernetes.container_name` is the name of a field in the log record, as described in the data model, see https://viaq.github.io/documentation/data_model/public/data_model.html#_kubernetes_container_name. Setting this as the structuredTypeKey means that the forwarder will examine each log record and use the value of the container_name field (if present) as the basis of the index key. Same goes for kubernetes.namespace_name or other keys. In theory you can use any field of the record, in practice we should only encourage examples that make sense (container, k8s label etc.)

Reply:
Updated. Thanks.